stl/inc/random

```diff
@@ -617,12 +617,37 @@ public:
 
     void discard(unsigned long long _Nskip) noexcept /* strengthened */ {
         // discard _Nskip elements
-        auto _Temp = _Prev;
-        for (; 0 < _Nskip; --_Nskip) {
-            _Temp = _Next_linear_congruential_value<_Uint, _Ax, _Cx, _Mx>(_Temp);
-        }
+        if constexpr (_Cx == 0 && _Mx == 2147483647) {
+            // for minstd_rand and minstd_rand0 we can improve performance by
+            // performing fast exponentiation and avoiding constant divisions
+            auto _Temp = static_cast<unsigned long long>(_Prev);
+            auto _Mul  = static_cast<unsigned long long>(_Ax);
+
+            for (;;) {
+                if (_Nskip & 1) {
+                    _Temp = _Temp * _Mul;
+                    _Temp = (_Temp >> 31) + (_Temp & _Mx);
+                    _Temp = _Temp < _Mx ? _Temp : _Temp - _Mx;
+                }
+                if (_Nskip >>= 1) {

```

Comment:
No change requested: We generally avoid directly testing integers for non-zero-ness (except when working with bitmasks), and in modern code we generally avoid mixing side effects into conditions (although our older code did that quite often).

However, in this case, I am okay with the style as-is. Perhaps it's the near-symmetry of the `_Temp` and `_Mul` logic, but I feel that mechanically following our usual conventions would decrease clarity.